connector/client/client.go

```diff
@@ -109,29 +141,35 @@ func (client *ConnectorClient) Put(ctx context.Context, target url.URL, activity
 	if err != nil {
 		return err
 	}
-	return client.sendRequest(req, activity)
+	return client.sendRequestWithRespErrCheck(req)
 }
 
-func (client *ConnectorClient) sendRequest(req *http.Request, activity schema.Activity) error {
+func (client *ConnectorClient) sendRequestWithRespErrCheck(req *http.Request) error {
+	res, err := client.sendRequest(req)
+	if err != nil {
+		return customerror.HTTPError{
+			HtErr: err,
+		}
```

Comment:
maybe we should create a helper function something like
```
func newHTTPError(err error){
    return  customerror.HTTPError{
        HtErr: err,
    }
}
```

Reply:
That sounds like a good idea 👍 Will do it!

Reply:
@PrasadG193 done!